### PR TITLE
SLCR-1: Initial Notification Form Backend

### PR DESCRIPTION
**Pull request by leshanw**

### CODE REVIEW

File: backend/services/src/casl/casl-ability.factory.ts [hunk 1]
```diff
@@ -291,16 +286,23 @@ export class CaslAbilityFactory {
...
+    ) {
+      can(Action.Read, CreditAuditLog);
+      can(Action.Read, CreditAuditLogViewEntity);
+    } else {
+      cannot(Action.Read, CreditAuditLog);
+      cannot(Action.Read, CreditAuditLogViewEntity);
+    }
+
+    if (user.companyRole === CompanyRole.PROGRAMME_DEVELOPER) {
+      can(Action.Read, ProgrammeSl);
```
Comment (tharindugayanga) [inline backend/services/src/casl/casl-ability.factory.ts:301]: Add SLCF and executive committee to view programmeSl as well

File: backend/services/src/programme-sl/programme-sl.service.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      return response;
+    } else {
+      throw new HttpException(
+        this.helperService.formatReqMessagesString("programme.docUploadFailed", []),
+        HttpStatus.INTERNAL_SERVER_ERROR
+      );
+    }
+  }
+
+  async create(programmeSlDto: ProgrammeSlDto, user: User): Promise<ProgrammeSl | undefined> {
```
Comment (tharindugayanga) [inline backend/services/src/programme-sl/programme-sl.service.ts:114]: Shall we keep the CRUD method order here? Create query update methods in that order. other methods can be private and at the bottom of the file as supporter methods

File: backend/services/src/casl/casl-ability.factory.ts (hunk 1 above)
Comment (leshanw) [reply backend/services/src/casl/casl-ability.factory.ts:301]: done

File: backend/services/src/programme-sl/programme-sl.service.ts (hunk 2 above)
Comment (leshanw) [reply backend/services/src/programme-sl/programme-sl.service.ts]: done

**Pull request merged**